driver-core/src/main/com/mongodb/internal/operation/ListCollectionsOperation.java

```diff
@@ -262,22 +262,22 @@ public void call(final AsyncConnectionSource source, final AsyncConnection conne
                 if (t != null) {
                     errHandlingCallback.onResult(null, t);
                 } else {
-                    final SingleResultCallback<AsyncBatchCursor<T>> wrappedCallback = releasingCallback(errHandlingCallback,
-                                                                                                        source, connection);
                     if (serverIsAtLeastVersionThreeDotZero(connection.getDescription())) {
-                        executeCommandAsync(binding, databaseName, getCommandCreator(), createCommandDecoder(),
-                                asyncTransformer(), retryReads,
+                        executeCommandAsyncWithConnection(binding, source, databaseName, getCommandCreator(), createCommandDecoder(),
+                                asyncTransformer(), retryReads, connection,
```

Comment:
Using the same connection to execute the command, as is done in other operations (and in the execute method)